### PR TITLE
[DOCS] Add output.console to Functionbeat doc and Functionbeat reference file

### LINKED ISSUE #16855: Document (docs and reference yml) new output.console for functionbeat
We [added](https://github.com/elastic/beats/pull/15143) `output.console` to help users debug in 7.6.0.  But this is not documented.  While this new output mode is intended for troubleshooting/debugging only, it can still benefit developers/admins for us to document it.

Proposal:
1.  Add documentation on this new output to https://www.elastic.co/guide/en/beats/functionbeat/current/configuring-output.html (similar to how we document console output for other beats)
2. Add a comment to `functionbeat.reference.yml` so developers know that such console output configuration exists if they just look at the reference yml

thx!

### PR DESCRIPTION
**Pull request by EamonnTP**
## What does this PR do?

This PR adds the `output.console` content to the Functionbeat doc and to the `functionbeat.reference.yml` file. 

## Why is it important?

`output.console` was added in 7.6.0 to help users debug, but it was not documented. While this new output mode is intended for troubleshooting/debugging only, it can still benefit developers/admins. The `functionbeat.reference.yml` file is also updated so developers know that the console output configuration exists.

## Related issues

Closes #16855

## HTML output

Coming soon. Doc build is currently failing and requires further investigation.

### CODE REVIEW

**Comment by bmorelli25** [comment]
@EamonnTP Merge this branch with master and push to retry the unsuccessful checks

**Comment by bmorelli25** [comment]
```
[2020-06-09T20:25:50.292Z] diff --git a/x-pack/functionbeat/functionbeat.reference.yml b/x-pack/functionbeat/functionbeat.reference.yml
[2020-06-09T20:25:50.292Z] index 0a3d6c5..61cf8a2 100644
[2020-06-09T20:25:50.292Z] --- a/x-pack/functionbeat/functionbeat.reference.yml
[2020-06-09T20:25:50.292Z] +++ b/x-pack/functionbeat/functionbeat.reference.yml
[2020-06-09T20:25:50.292Z] @@ -927,19 +927,9 @@ output.elasticsearch:
[2020-06-09T20:25:50.292Z]    # timing out. The default is 30s.
[2020-06-09T20:25:50.292Z]    #timeout: 30s
[2020-06-09T20:25:50.292Z]  
[2020-06-09T20:25:50.292Z] -# =============================== Console Output ===============================
[2020-06-09T20:25:50.292Z]  
[2020-06-09T20:25:50.292Z] -#output.console:
[2020-06-09T20:25:50.292Z] -  # Boolean flag to enable or disable the output module.
[2020-06-09T20:25:50.292Z] -  #enabled: true
[2020-06-09T20:25:50.292Z]  
[2020-06-09T20:25:50.292Z] -  # Configure JSON encoding
[2020-06-09T20:25:50.292Z] -  #codec.json:
[2020-06-09T20:25:50.292Z] -    # Pretty-print JSON event
[2020-06-09T20:25:50.292Z] -    #pretty: false
[2020-06-09T20:25:50.292Z]  
[2020-06-09T20:25:50.292Z] -    # Configure escaping HTML symbols in strings.
[2020-06-09T20:25:50.292Z] -    #escape_html: false
[2020-06-09T20:25:50.292Z]  
[2020-06-09T20:25:50.292Z]  # =================================== Paths ====================================
[2020-06-09T20:25:50.292Z]  
[2020-06-09T20:25:50.292Z] x-pack/functionbeat/functionbeat.reference.yml: needs update
```

**Comment by dedemorton** [comment]
Aha. So I should have caught this during the review, but it's been awhile since I've updated the config reference, and I'd forgotten that they're now using templating to create the reference files. 

So instead of manually updating the functionbeat.reference.yml, you need to make a code change.

https://github.com/elastic/beats/blob/master/x-pack/functionbeat/scripts/mage/config.go#L16

Then run `make update` from the `beats` directory. Note that you have to have your dev environment set up correctly to run this command.

**Comment by dedemorton** [comment]
(or I guess you could set it to `false` but I don't see it set anywhere else, so removing probably makes more sense).

**Comment by ppf2** [comment]
One thing we might want to add (open for discussion) is a warning about `output.console` to be used for debugging only for it can produce a large amounts of logging (this can translate to $$ if the output of the logging ends up in Cloudwatch logs).  Not sure if we need to explicitly call this out though or if it's implied :)

File: libbeat/outputs/console/docs/console.asciidoc [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 The Console output writes events in JSON format to stdout.
```
Comment (urso) [inline libbeat/outputs/console/docs/console.asciidoc:8]: One can configure a custom formatter to print messages with a custom layout. JSON is just the default formatter (see codec setting).

**Pull request merged**